### PR TITLE
Docs(merge_strategies): Clarfiy rebase and ff merge similarity

### LINKED ISSUE #1387: Description of rebase merge method is incorrect
### What article on docs.github.com is affected?

https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/about-merge-methods-on-github

Contained in:
https://github.com/github/docs/blob/main/content/github/administering-a-repository/about-merge-methods-on-github.md

Which reads from:
https://github.com/github/docs/blob/main/data/reusables/pull_requests/rebase_and_merge_summary.md

### What part(s) of the article would you like to see updated?

In the section `Rebasing and merging your commits`, the description says
> Pull requests with rebased commits are merged using the fast-forward option.

However, this is not the case. Rebasing is actually done with the `--force` option as described later in the section where it clarifies that the commit hash and the committer will change. This is explicitly forced rebase behaviour and quite different from fast-forward, which leaves all commits as-is.

### Additional information

I consider this confusing, as someone looking for a fast-forward merge option might stop at the end of the intro, thinking they have found the option they need, but in fact will get different behaviour.

### PR DESCRIPTION
**Pull request by devtayls**
### Why:

Closes [1387](https://github.com/github/docs/issues/1387)

### What's being changed:

**Content Changes**
> Pull requests with rebased commits are merged using the [fast-forward option](https://git-scm.com/docs/git-merge#_fast_forward_merge).
 
to: 

>In that way, the rebase and merge behavior resembles a [fast-forward merge](https://git-scm.com/docs/git-merge#_fast_forward_merge) by maintaining a linear project history. However, rebasing achieves this by re-writing the commit history on the base branch with new commits.

This is because as @bk2204, @spikyjt explained in the issue, rebasing _looks_ like `ff-merge`, but in actuality is a destructive method which changes the project history.

**Refactor**

Move paragraph on enabling **Rebase and Merge** behind paragraph explaining how rebasing works on Github. This flows better as: 'What is rebase and merge?' -> 'How do I enable Rebase and Merge?' Than the previous version which was: 'What is rebase and merge?' -> 'How do I enable Rebase and Merge' -> 'How github rebase differs from git rebase'

### Check off the following:

- [ ] I have reviewed my changes in staging (look for the latest deployment event in your pull request's timeline, then click **View deployment**).
- [ ] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

File: data/reusables/pull_requests/rebase_and_merge_summary.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
...
 
-To rebase and merge pull requests, you must have [write permissions](/articles/repository-permission-levels-for-an-organization/) in the repository, and the repository must [allow rebase merging](/articles/configuring-commit-rebasing-for-pull-requests/).
+The rebase and Merge behavior on {% data variables.product.product_name %} deviates slightly from `git rebase`. Rebase and merge on {% data variables.product.prodname_dotcom %} will always update the committer information and create new commit SHAs, whereas `git rebase` outside of {% data variables.product.prodname_dotcom %} does not change the committer information when the rebase happens on top of an ancestor commit. For more information about `git rebase`, see [the official Git documentation](https://git-scm.com/docs/git-rebase).
```
Comment (lecoursen) [inline data/reusables/pull_requests/rebase_and_merge_summary.md:3]:
```suggestion
The rebase and merge behavior on {% data variables.product.product_name %} deviates slightly from `git rebase`. Rebase and merge on {% data variables.product.prodname_dotcom %} will always update the committer information and create new commit SHAs, whereas `git rebase` outside of {% data variables.product.prodname_dotcom %} does not change the committer information when the rebase happens on top of an ancestor commit. For more information about `git rebase`, see [git-rebase](https://git-scm.com/docs/git-rebase) in the Git documentation.
```

**Review by lecoursen — COMMENTED** [review]
Thanks @devtayls, this looks great! I'm applying a suggestion with a few small style tweaks and will merge after CI finishes! ⚡

**Review by lecoursen — APPROVED** [review]

**Comment by lecoursen** [comment]
The link check errors are unrelated to this PR.

File: data/reusables/pull_requests/rebase_and_merge_summary.md (hunk 1 above)
Comment (devtayls) [reply data/reusables/pull_requests/rebase_and_merge_summary.md]: coolcoolcool. 

Much better. :)

**Pull request merged**